simulation/simple_sensor_simulator/test/src/sensor_simulation/detection_sensor/test_detection_sensor.cpp

```diff
@@ -0,0 +1,316 @@
+ * Ego.
+ */
+TEST_F(DetectionSensorTest, update_motorcycleSubtype)
+{
+  initializeEntityStatus("motorcycle", EntityType::VEHICLE, EntitySubtype::MOTORCYCLE);
+
+  detection_sensor_->update(
+    current_simulation_time_, status_, current_ros_time_, lidar_detected_entities_);
+
+  // Spin the node to process callbacks
+  rclcpp::spin_some(node_);
+
+  ASSERT_NE(received_msg_, nullptr);
+  ASSERT_EQ(received_msg_->objects.size(), static_cast<size_t>(1));
+  EXPECT_EQ(received_msg_->objects[0].classification[0].label, ObjectClassification::MOTORCYCLE);
+  EXPECT_POSITION_NEAR(
+    received_msg_->objects[0].kinematics.pose_with_covariance.pose.position, entity_pose_.position,
+    std::numeric_limits<double>::epsilon());
+}
+
+/**
+ * @note Test basic functionality. Test update process correctness with no position noise, no delay
+ * and filter_by_range = true (in configuration), no probability of lost (configuration) and BICYCLE
+ * entity positioned closer to Ego than the range parameter (in configuration) - the goal is to test
+ * the standard detection functionality when an entity is detected because it is close to Ego
+ */
+TEST_F(DetectionSensorTest, update_bicycleSubtype)
+{
+  initializeEntityStatus("bicycle", EntityType::VEHICLE, EntitySubtype::BICYCLE);
+
+  detection_sensor_->update(
+    current_simulation_time_, status_, current_ros_time_, lidar_detected_entities_);
+
+  // Spin the node to process callbacks
+  rclcpp::spin_some(node_);
+
+  ASSERT_NE(received_msg_, nullptr);
+  ASSERT_EQ(received_msg_->objects.size(), static_cast<size_t>(1));
+  EXPECT_EQ(received_msg_->objects[0].classification[0].label, ObjectClassification::BICYCLE);
+  EXPECT_POSITION_NEAR(
+    received_msg_->objects[0].kinematics.pose_with_covariance.pose.position, entity_pose_.position,
+    std::numeric_limits<double>::epsilon());
+}
+
+/**
+ * @note Test basic functionality. Test update process correctness with no position noise, no delay
+ * and filter_by_range = true (in configuration), no probability of lost (configuration) and
+ * PEDESTRIAN entity positioned closer to Ego than the range parameter (in configuration) - the goal
+ * is to test the standard detection functionality when an entity is detected because it is close to
+ * Ego
+ */
+TEST_F(DetectionSensorTest, update_pedestrianSubtype)
+{
+  initializeEntityStatus("pedestrian", EntityType::PEDESTRIAN, EntitySubtype::PEDESTRIAN);
+
+  detection_sensor_->update(
+    current_simulation_time_, status_, current_ros_time_, lidar_detected_entities_);
+
+  // Spin the node to process callbacks
+  rclcpp::spin_some(node_);
+
+  ASSERT_NE(received_msg_, nullptr);
+  ASSERT_EQ(received_msg_->objects.size(), static_cast<size_t>(1));
+  EXPECT_EQ(received_msg_->objects[0].classification[0].label, ObjectClassification::PEDESTRIAN);
+  EXPECT_POSITION_NEAR(
+    received_msg_->objects[0].kinematics.pose_with_covariance.pose.position, entity_pose_.position,
+    std::numeric_limits<double>::epsilon());
+}
```

Comment:
you could look into templated tests or write a simple macro. It would certainly reduce the amount of code

Reply:
Ok, it done. Thanks!